src/nodes/display/PassNode.js

```diff
@@ -157,47 +157,50 @@ class PassNode extends TempNode {
 
 	getTextureNode( name = 'output' ) {
 
-		let textureNode = this._nodes[ name ];
+		let textureNode = this._textureNodes[ name ];
 
 		if ( textureNode === undefined ) {
 
-			this._nodes[ name ] = textureNode = nodeObject( new PassMultipleTextureNode( this, name ) );
+			this._textureNodes[ name ] = textureNode = nodeObject( new PassMultipleTextureNode( this, name ) );
 
 		}
 
 		return textureNode;
 
 	}
 
-	getViewZNode() {
+	getViewZNode( name = 'depth' ) {
 
-		if ( this._viewZNode === null ) {
+		let viewZNode = this._viewZNodes[ name ];
+
+		if ( viewZNode === undefined ) {
 
 			const cameraNear = this._cameraNear;
 			const cameraFar = this._cameraFar;
 
-			this._viewZNode = perspectiveDepthToViewZ( this.getTextureNode( 'depth' ), cameraNear, cameraFar );
+			this._viewZNodes[ name ] = viewZNode = perspectiveDepthToViewZ( this.getTextureNode( name ), cameraNear, cameraFar );
 
 		}
 
-		return this._viewZNode;
+		return viewZNode;
 
 	}
 
-	getLinearDepthNode() {
+	getLinearDepthNode( name = 'depth' ) {
+
+		let linearDepthNode = this._linearDepthNodes[ name ];
 
-		if ( this._linearDepthNode === null ) {
+		if ( linearDepthNode === undefined ) {
 
 			const cameraNear = this._cameraNear;
 			const cameraFar = this._cameraFar;
+			const viewZNode = this.getViewZNode( name );
 
-			// TODO: just if ( builder.camera.isPerspectiveCamera )
```

Comment:
It would be interesting to preserve the TODO, this should be implemented at some point.

Reply:
Yes, that should be left in.